### PR TITLE
(fix) (773) Dependent field validator more resilient

### PR DESCRIPTION
**Pull request by CristinaRO**

### CODE REVIEW

**Comment by CristinaRO** [comment]
The failures are related to a different area of the code, which has a fix in `develop`: 60629efa 🤔

File: app/validators/dependent_field_inclusion_validator.rb [hunk 1]
```diff
@@ -2,11 +2,14 @@ class DependentFieldInclusionValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
     parent_field_name = options[:parent]
     parent_field_value = record.attributes[parent_field_name]
-    parent_field_value_lookup = parent_field_value.to_s.downcase
-    mapping = options[:in]
 
-    valid_values = mapping.dig(parent_field_name, parent_field_value_lookup).map(&:downcase)
-    return if value.downcase.in?(valid_values)
+    unless parent_field_value.nil?
```
Comment (rgarner) [inline app/validators/dependent_field_inclusion_validator.rb]: I think Rubocop normally complains about `unless` in a multiline construct as a large chunk of code can be predicated on a double negative.

```suggestion
    if parent_field_value
```

File: spec/validators/dependent_field_inclusion_validator_spec.rb [hunk 2]
```diff
@@ -63,4 +63,28 @@ def self.name
...
+    let(:data) { { 'Service Type' => 'Core', 'Primary Specialism' => nil } }
+
+    it { is_expected.to_not be_valid }
+  end
+
+  context 'the primary specialism field is missing' do
+    let(:data) { { 'Service Type' => 'Core' } }
+
+    it { is_expected.to_not be_valid }
+  end
```
Comment (rgarner) [inline spec/validators/dependent_field_inclusion_validator_spec.rb]: There's only a need for both these specs (this and the explicit `=> nil` on L68) if we're using `fetch` behaviour, but we're using `dig` in the validator which can return nil without an exception, so I think one of these is redundant.

Comment (CristinaRO) [reply spec/validators/dependent_field_inclusion_validator_spec.rb]: We can poke at this tomorrow in more leisure, as we're not treating it as quite a hotfix anymore.

File: spec/validators/dependent_field_inclusion_validator_spec.rb [hunk 3]
```diff
@@ -85,6 +85,12 @@ def self.name
     it { is_expected.to_not be_valid }
   end
 
+  context 'the parent field has a value not present in our mapping' do
+    let(:data) { { 'Service Type' => 'bogus', 'Primary Specialism' => 'Equity Capital Markets' } }
```
Comment (rgarner) [inline spec/validators/dependent_field_inclusion_validator_spec.rb]: :100: intent really clear

**Pull request merged**